### PR TITLE
Enable default service for GAE

### PR DESCRIPTION
**Pull request by psx95**
This PR adds terraform resources to create and deploy the default service for Google App Engine. 

### Testing
 - Ran the `apply-persistent` script & verified that the default service was created. 
 - As explained in the comments in the added `Dockerfile`, the container for default service was already uploaded to the Artifact Registry. 

#### NOTE: GAE Application 
 - Due to the known GCP limitations where the GAE application cannot be deleted once deployed, the terraform script would always fail with error after the initial run - 
      ```
       Error: Error creating App Engine application: googleapi: Error 409: This application already exists and cannot be re-created.
      ```
For this reason, this part was moved out of the script, since the `tf/persistent` can be updated later. 

The GAE app creation is really needed only once in the lifecycle of a GCP project and manual steps involve only selecting - 
 1. service account
 2. region

Creating GAE Application should therefore be added to the documentation in case the project is ever re-created under a different name in GCP.

### CODE REVIEW

**Comment by dashpole** [comment]
> Creating GAE Application should therefore be added to the documentation in case the project is ever re-created under a different name in GCP.

I'm not sure it makes sense to add this to the repo in that case.  If we were to run this a few years from now, it might not work?  It is basically untested code.  I would be in favor of running it once, but not committing the code.  If we ever need to run it again, we could reference this PR, which documents what we did last time.

**Comment by psx95** [comment]
> > Creating GAE Application should therefore be added to the documentation in case the project is ever re-created under a different name in GCP.
> 
> I'm not sure it makes sense to add this to the repo in that case. If we were to run this a few years from now, it might not work? It is basically untested code. I would be in favor of running it once, but not committing the code. If we ever need to run it again, we could reference this PR, which documents what we did last time.

#### Clarification on the PR description. 
So, there are 2 parts to GAE - creating the GAE Application (which is what I am talking about in the PR description & it is not included in this code) and then creating the GAE service that runs within the application. 

Some context on GAE - 
 - There can only be 1 GAE application per project, but that application can contain multiple services.
 - The GAE application once created cannot be deleted (you can disable it though).
 - There is a "default" service which is required before you can create any other service. This service cannot be deleted.

The above code automates the process of creating & configuring the default service. This code is tested and if we run this code in a few years from now (to bootstrap a new project) it should work, provided that we - 
 - Update the documentation to add steps to enable the Google App Engine API and create the GAE application before running the `apply-persistent` script. 

#### Why App Creation (not default service) was not automated - 

Since GAE application cannot be deleted and there was no easy way to skip resource creation in case the resource is already created, [google_app_engine_application](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/app_engine_application), the `apply-persistent` script reported failure every time it ran (**it doesn't now, since this google_app_engine_application resource was removed**)  since the application was already created. 

NOTE: A frequent ask in the open source community was to skip the resource creation instead of reporting failure, so it maybe added in terraform in future.

Let me know if this makes it clear.

**Comment by dashpole** [comment]
lets chat about it

**Review by dashpole — APPROVED** [review]
We already have other setup scripts/terraform in this repo.  This keeps that consistent.

We can reevaluate later if it useful to keep single-time setup code/scripts/config around, but let's keep it for now.

**Comment by psx95** [comment]
Will be adding changes to the README in this folder before merging.

README will be updated to include information on how the terraform files in this folder (tf/persistent) should be used.

**Review by dashpole — APPROVED** [review]

File: gae-service-container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# This docker image should currently be built manually and pushed to a proper location in the artifact repository
+# Docker image name - us-central1-docker.pkg.dev/opentelemetry-ops-e2e/gae-service-containers/default-service:latest
+# TODO: Automate this process
+
+FROM httpd:2.4
+
+# This configuration file is extracted from the above image and has been modified so that the server listens to port 8080 instead of port 80.
+# The steps to do this are described here - https://hub.docker.com/_/httpd
+COPY ./httpd.conf /usr/local/apache2/conf/httpd.conf
```
Comment (aabmass) [inline gae-service-container/Dockerfile:27]: I don't want to block things but apache seems a bit overkill for this and the included httpd.conf is pretty bulky. Maybe one of these options would be easier:

- copy this sort of basic code https://cloud.google.com/appengine/docs/flexible/go/create-app#helloworldgo to and build an image with it
- `docker run --rm -p 8080:8080 python:3-alpine python -m http.server 8080`?

File: tf/persistent/README.md [hunk 2]
```diff
@@ -1,7 +1,118 @@
...
-Right now it manages:
+This module is for creating resources that should be persistent across tests. There are no automated tests for the terraform files in this directory. 
 
-- The GKE cluster for running test pods
-- The Cloud Build CI Triggers for each repo
+Most of the resources used by these tests are set up and torn down for each test run. There are however some persistent resources which are managed by terraform (under [tf/persistent](../persistent/)) in the default terraform workspace:
+
+- The GKE cluster ([e2etest-default](https://pantheon.corp.google.com/kubernetes/clusters/details/us-central1/e2etest-default/details?project=opentelemetry-ops-e2e)) for running test pods.
+- The Cloud Build CI Triggers for each repo ([pantheon link](https://pantheon.corp.google.com/cloud-build/triggers?project=opentelemetry-ops-e2e)).
+- The default service for the created GAE application ([pantheon link](https://pantheon.corp.google.com/appengine/services?serviceId=default&versionId=v1&project=opentelemetry-ops-e2e)).
```
Comment (aabmass) [inline tf/persistent/README.md:9]: I don't think we want to link to our actual project in these docs, or if we do it shouldn't be a `*.corp.google.com` link

File: tf/persistent/README.md [hunk 3]
```diff
@@ -1,7 +1,118 @@
...
+    "$@"
+}
+opentelemetry_operations_e2e_testing apply-persistent
+```
+
+and then executing the script using - 
+
+```
+sh <your_script_name>
+```
```
Comment (aabmass) [inline tf/persistent/README.md:51]: why not just paste the above into your shell instead of saving a script?

Comment (psx95) [reply tf/persistent/README.md:51]: Good point, will do that.

**Pull request merged**